nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutTCP.java

```diff
@@ -0,0 +1,290 @@
+            if (!isConnectionPerFlowFile(context)) {
+                relinquishSender(sender);
+            } else {
+                sender.close();
+            }
+        }
+    }
+
+    /**
+     * event handler method to perform the required actions when a failure has occurred. The FlowFile is penalized, forwarded to the failure relationship and the context is yielded.
+     *
+     * @param context
+     *            - the current process context.
+     *
+     * @param session
+     *            - the current process session.
+     * @param flowFile
+     *            - the FlowFile that has failed to have been processed.
+     */
+    protected void onFailure(final ProcessContext context, final ProcessSession session, final FlowFile flowFile) {
+        session.transfer(session.penalize(flowFile), REL_FAILURE);
+        session.commit();
+        context.yield();
+    }
+
+    /**
+     * Helper method to read the FlowFile content stream into a ByteArrayOutputStream object.
+     *
+     * @param session
+     *            - the current process session.
+     * @param flowFile
+     *            - the FlowFile to read the content from.
+     *
+     * @return ByteArrayOutputStream object containing the FlowFile content.
+     */
+    protected ByteArrayOutputStream readContent(final ProcessSession session, final FlowFile flowFile) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream((int) flowFile.getSize() + 1);
+        session.read(flowFile, new InputStreamCallback() {
+            @Override
+            public void process(final InputStream in) throws IOException {
+                StreamUtils.copy(in, baos);
+            }
+        });
+
+        return baos;
+    }
+
+    /**
+     * Helper method to append a delimiter to the message contents.
+     *
+     * @param content
+     *            - the message contents.
+     * @param delimiter
+     *            - the delimiter value.
+     *
+     * @return ByteArrayOutputStream object containing the new message contents.
+     */
+    protected ByteArrayOutputStream appendDelimiter(final ByteArrayOutputStream content, final String delimiter, Charset charSet) {
+        content.write(delimiter.getBytes(charSet), 0, delimiter.length());
+        return content;
+    }
+
+    /**
+     * Gets the current value of the "Outgoing Message Delimiter" property.
+     *
+     * @param context
+     *            - the current process context.
+     * @param flowFile
+     *            - the FlowFile being processed.
+     *
+     * @return String containing the Delimiter value.
+     */
+    protected String getDelimiter(final ProcessContext context, final FlowFile flowFile) {
+        String delimiter = context.getProperty(OUTGOING_MESSAGE_DELIMITER).evaluateAttributeExpressions(flowFile).getValue();
+        if (delimiter != null) {
+            delimiter = delimiter.replace("\\n", "\n").replace("\\r", "\r").replace("\\t", "\t");
```

Comment:
@joewitt I would definitely be on board for find a better way to do it through the UI but this isn't entirely inconsistent with how it's already done. In fact, scanning for "\\n" in the code base I found multiple instances where "message delimiters/demarcators" replace "\\n" with "\n"[1][2][3]. A more telling point of view is to look for all those properties where it has a default of a new line. The only 3 are those where they replace "\\n" with "\n". So from one point of view, yes we are inconsistent to have special places where "\n" in the UI is replace with a new line character but we are consistently inconsistent such that all those that do regularly use just new lines replace "\\n" with "\n".

[1] https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/GetKafka.java#L401-L401
[2] https://github.com/apache/nifi/blob/master/nifi-commons/nifi-processor-utilities/src/main/java/org/apache/nifi/processor/util/listen/AbstractListenEventBatchingProcessor.java#L97-L97
[3] https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenSyslog.java#L268-L268